src/login/LoginPage.jsx

```diff
@@ -1,435 +1,266 @@
+  /**
+   * Backup the login form in redux when login page is toggled.
+   */
+  useEffect(() => {
+    if (shouldBackupState) {
+      backupFormState({
+        formFields: { ...formFields },
+        errors: { ...errors },
+      });
     }
-    this.props.getThirdPartyAuthContext(payload);
-    this.props.loginRequestReset();
-  }
+  }, [shouldBackupState, formFields, errors, backupFormState]);
 
-  shouldComponentUpdate(nextProps) {
-    if (nextProps.loginFormData && this.props.loginFormData !== nextProps.loginFormData) {
-      // Ensuring browser's autofill user credentials get filled and their state persists in the redux store.
-      const nextState = {
-        emailOrUsername: nextProps.loginFormData.emailOrUsername || this.state.emailOrUsername,
-        password: nextProps.loginFormData.password || this.state.password,
-      };
-      this.setState({
-        ...nextProps.loginFormData,
-        ...nextState,
-      });
-      return false;
+  useEffect(() => {
+    if (loginErrorCode) {
+      setErrorCode(prevState => ({ type: loginErrorCode, count: prevState.count + 1 }));
     }
-    return true;
-  }
+  }, [loginErrorCode]);
 
-  componentWillUnmount() {
-    if (this.props.resetPassword) {
-      this.props.loginRemovePasswordResetBanner();
+  useEffect(() => {
+    if (loginResult.success) {
+      // TODO: Do we still need this cookie?
+      setSurveyCookie('login');
     }
-  }
+  }, [loginResult]);
 
-  getEnterPriseLoginURL() {
-    return getConfig().LMS_BASE_URL + ENTERPRISE_LOGIN_URL;
-  }
+  const validateFormFields = (payload) => {
+    const { emailOrUsername, password } = payload;
+    const fieldErrors = { ...errors };
 
-  handleSubmit = (e) => {
-    e.preventDefault();
-    if (this.props.resetPassword) {
-      this.props.loginRemovePasswordResetBanner();
+    if (emailOrUsername === '') {
+      fieldErrors.emailOrUsername = formatMessage(messages['email.validation.message']);
+    } else if (emailOrUsername < 3) {
+      fieldErrors.emailOrUsername = formatMessage(messages['username.or.email.format.validation.less.chars.message']);
+    }
+    if (password === '') {
+      fieldErrors.password = formatMessage(messages['password.validation.message']);
     }
-    this.setState({ isSubmitted: true });
-    const { emailOrUsername, password } = this.state;
-    const emailValidationError = this.validateEmail(emailOrUsername);
-    const passwordValidationError = this.validatePassword(password);
 
-    if (emailValidationError !== '' || passwordValidationError !== '') {
-      this.props.setLoginFormData({
-        errors: {
-          emailOrUsername: emailValidationError,
-          password: passwordValidationError,
-        },
-      });
-      this.props.loginRequestFailure({
-        errorCode: INVALID_FORM,
-      });
+    return { ...fieldErrors };
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (showResetPasswordSuccessBanner) {
```

Comment:
If we are only dismissing the banner here. can we do it in the `loginRequest` action inside the reducer instead of creating a whole new action?

Reply:
Not all handleSubmit requests call `loginRequest`, some fail on the frontend validations and return before that.